src/resolvers/Query/helperFunctions/getInputArgs.ts

```diff
@@ -0,0 +1,692 @@
+import {
+  DonationWhereInput,
+  EventWhereInput,
+  InputMaybe,
+  OrganizationWhereInput,
+  PostWhereInput,
+  UserWhereInput,
+} from "../../../types/generatedGraphQLTypes";
+
+export const getInputArgs = (
+  where:
+    | InputMaybe<
+        EventWhereInput &
+          OrganizationWhereInput &
```

Comment:
If `|` is used here, we would need to add specific checks in each branch. For example,
```ts
  if (where.title_not) {
    inputArgsPayload = {
      ...inputArgsPayload,
      title: { $ne: where.title_not },
    };
  }
```
In this branch, typescript would thrown an error because `title_not` doesn’t exist on `OrganizationWhereInput`. So, an additional check before this branch would be required to assure that `where.title_not` is checked only after confirming that the passed argument doesn’t belong to `OrganizationWhereInput`.
`&` works here because the we know every property of `WhereInput`s will be optional, i.e. it can be undefined.
For example,
```ts
export type DonationWhereInput = {
  id?: InputMaybe<Scalars[‘ID’]>;
  id_contains?: InputMaybe<Scalars[‘ID’]>;
  id_in?: InputMaybe<Array<Scalars[‘ID’]>>;
  id_not?: InputMaybe<Scalars[‘ID’]>;
  id_not_in?: InputMaybe<Array<Scalars[‘ID’]>>;
  id_starts_with?: InputMaybe<Scalars[‘ID’]>;
  name_of_user?: InputMaybe<Scalars[‘String’]>;
  name_of_user_contains?: InputMaybe<Scalars[‘String’]>;
  name_of_user_in?: InputMaybe<Array<Scalars[‘String’]>>;
  name_of_user_not?: InputMaybe<Scalars[‘String’]>;
  name_of_user_not_in?: InputMaybe<Array<Scalars[‘String’]>>;
  name_of_user_starts_with?: InputMaybe<Scalars[‘String’]>;
};
```

Reply:
@beingnoble03 Hmm, understood. But your approach has a problem too, it's expecting that all fields in the filter object are optional. Though filter fields in most use cases would be optional, we can still make it much better. Even if the fields aren't optional, we can make them optional just to make sure that even if a required field exists in filter object, while passing it inside this utility function it will still be optional.

Do it like this:-
```ts
  where:
    | InputMaybe<
        Partial<
          EventWhereInput &
            OrganizationWhereInput &
            PostWhereInput &
            UserWhereInput &
            DonationWhereInput
        >
      >
    | undefined
```
By using `Partial` generic you make all the fields inside the big object created using `&` type optional.

Other than that you still have to manually defined the shape of of objects this function will return for different inputs. But since you've merged all the objects together using `&`, you'll have to use generics for it.

This is how it needs to be done:-
```ts
export function getInputArgs<T>(...): {
  ...
  return inputArgsPayload as T;
}
```

When you call this function from a resolver, you'll have to pass in the custom filter object shape(type) that you expect from this function using generic type `T`. This way you'll have a type-safe filter object returned from this function.

Reply:
Done.